src/Mod/Draft/draftfunctions/scale.py

```diff
@@ -71,11 +71,9 @@ def scale(objectslist, scale=App.Vector(1,1,1),
         if hasattr(obj,'Shape'):
             scaled_shape = obj.Shape.copy()
             m = App.Matrix()
-            m.move(obj.Placement.Base.negative())
             m.move(center.negative())
             m.scale(scale.x,scale.y,scale.z)
             m.move(center)
-            m.move(obj.Placement.Base)
             scaled_shape = scaled_shape.transformGeometry(m)
```

Comment:
Does the command still work good with simple Part::Feature objects if we delete those lines?

Reply:
The code we are dealing with is inside the 'elif tp in ["Rectangle"]:' block. So simple Part::Feature objects should not be affected by the suggested modifications.

Reply:
Isn't it inside the `if hasattr(obj,'Shape'):`? Looks like the `if utils.get_type(obj) == "Rectangle"` comes after...

Reply:
You are right, I was mixing up two files. Although the Placement.Base vector moves seem illogical, I'll check this.

Reply:
Thanks :)
Consider it could be like this to correctly apply the transformation matrix for Part::Features.

Reply:
The original code does not work for simple Part::Feature objects. Which is a bit confusing as a correct preview is shown when the X, Y and Z scales are entered.

Error message:
Unable to scale object: Shape - This object type cannot be scaled directly. Please use the clone method.

Shape was created with the code below. Alternatively you can create a Part_SimpleCopy from a Draft rectangle.
```
import Part
import FreeCAD as App

V1 = App.Vector(0, 10, 0)
V2 = App.Vector(30, 10, 0)
V3 = App.Vector(30, -10, 0)
V4 = App.Vector(0, -10, 0)

L1 = Part.LineSegment(V1, V2)
L2 = Part.LineSegment(V2, V3)
L3 = Part.LineSegment(V3, V4)
L4 = Part.LineSegment(V4, V1)

S1 = Part.Shape([L1, L2, L3, L4])

W = Part.Wire(S1.Edges)
Part.show(W)
```

Reply:
Strange, this function appears to be exactly made to scale Part::Features. 
your error come from [here](https://github.com/FreeCAD/FreeCAD/blob/33ef4e0e35fba74c6e2fa3829cfd15c4b6f67e94/src/Mod/Draft/draftguitools/gui_scale.py#L329), because the Part::Feature is considered [non scalable:](https://github.com/FreeCAD/FreeCAD/blob/33ef4e0e35fba74c6e2fa3829cfd15c4b6f67e94/src/Mod/Draft/draftguitools/gui_scale.py#L296).
But that's strange, because the function should be able to do it (reading the code). :)

Reply:
The reason is perhaps that a simple Part::Feature can contain arcs and circles? There are other things that need addressing in this file. For example: unequally scaling a rotated rectangle should result in a wire IMO. Comparable to what happens when you stretch one point of a rectangle. I suggest merging this PR as an interim improvement.

Reply:
As advised by  vocx-fc I have check the Arch and BIM workbenches. Draft.scale is not used there.

Reply:
@carlopav 
You are right: the scale function in scale.py works on simple Part::Features. In my previous test I have used the Draft_Scale command (as you already pointed out). I'll do some testing and then report back.

Reply:
I have now tested with 4 object types. See below. Are there other object types I should test with? I do not know how to create an object of the "Annotation" type. See this forum topic: https://forum.freecadweb.org/viewtopic.php?f=23&t=49942.

Tested object types:
1. Rectangles: This PR is required.
2. Wires: This PR not required, but the PR does not cause problems.
3. Simple Part::Features: This PR is required.
4. Spline: This PR not required, but the PR does not cause problems.

Test code for Simple Part::Features:
```
import FreeCAD as App
import Draft
import Part

cen = App.Vector(2, 1, 0)   # center for scaling and base point of feature
pts = [App.Vector(0, 0, 0),
       App.Vector(4, 0, 0),
       App.Vector(4, 3, 0),
       App.Vector(0, 3, 0)]
shp = Part.Shape([Part.LineSegment(pts[0], pts[1]),
                  Part.LineSegment(pts[1], pts[2]),
                  Part.LineSegment(pts[2], pts[3]),
                  Part.LineSegment(pts[3], pts[0])])
obj = App.ActiveDocument.addObject("Part::Feature")
obj.Shape = shp
obj.Placement.Base = cen # must happen after changing the Shape property
App.ActiveDocument.recompute()

sca = App.Vector(2, 3, 1)
Draft.scale([obj], sca, cen, True)
App.ActiveDocument.recompute()
```

Test code for splines:
```
import FreeCAD as App
import Draft
import Part

cen = App.Vector(2, 1, 0)   # center for scaling and base point of spline
pts = [App.Vector(0, 0, 0),
       App.Vector(2, 3, 0),
       App.Vector(4, 0, 0)]
obj = Draft.make_bspline(pts, False)
obj.Placement.Base = cen
App.ActiveDocument.recompute()

sca = App.Vector(2, 3, 1)
Draft.scale([obj], sca, cen, True)
App.ActiveDocument.recompute()
```

Reply:
Very good! thanks for testing :)

Reply:
FYI: Draft.scale can scale simple Part::Features that contain arcs. But because a Part::Feature is a more or less generic container, gui_scale.py probably rejects them. Analyzing their content before processing is perhaps difficult?

Example: A simple copy of a Part cylinder can be Draft.scaled. But this takes quite a long time, suggesting that the result contains very complex geometry.

Reply:
> Strange, this function appears to be exactly made to scale Part::Features.
> your error come from [here](https://github.com/FreeCAD/FreeCAD/blob/33ef4e0e35fba74c6e2fa3829cfd15c4b6f67e94/src/Mod/Draft/draftguitools/gui_scale.py#L329), because the Part::Feature is considered [non scalable:](https://github.com/FreeCAD/FreeCAD/blob/33ef4e0e35fba74c6e2fa3829cfd15c4b6f67e94/src/Mod/Draft/draftguitools/gui_scale.py#L296).
> But that's strange, because the function should be able to do it (reading the code). :)

Any shape which is attached to an object is immutable. So in order to change the shape of a Part Feature, you need to make a copy of it, modify the copy, then attribute back the copy to the shape:
```
shp = obj.Shape.copy()
# .. modify shp...
obj.Shape = shp
```

Reply:
@yorikvanhavre This is exactly what happens in scale.py.

Reply:
@yorikvanhavre Thanks for merging.

Reply:
I have prepared some unit tests for scale.py. I'll do a separate PR for them.